### PR TITLE
Support namespace ExtendedOptions in cluster spec

### PR DESCRIPTION
**Pull request by linasm**
This adds support for namespace `ExtendedOptions` field in cluster spec map (passing its value to namespace create request, if present).

Note that I have used intermediary transformation via json (from cluster spec to protobuf) to avoid implementing manual, error prone conversion of dynamic `map[string]interface{}` which is used in `ExtendedOptions`.

### CODE REVIEW

File: pkg/apis/m3dboperator/v1alpha1/namespace.go [hunk 1]
```diff
@@ -98,6 +100,37 @@ type DownsampleOptions struct {
...
+}
+
+func (o *dynamicOption) MarshalJSON() ([]byte, error) {
+	return json.Marshal(o.o)
+}
+
+// ExtendedOptions stores the extended namespace options.
+type ExtendedOptions struct {
+	Type    string                   `json:"type,omitempty"`
+	Options map[string]DynamicOption `json:"options,omitempty"`
```
Comment (schallert) [inline pkg/apis/m3dboperator/v1alpha1/namespace.go:131]: I've never seen interface types in Kubernetes API types before. I wonder if having this be `map[string]json.RawMessage` would give you the same benefits (auto-generated DeepCopy implementation, more control over serialization) while being more Kubernetes-friendly?

Comment (linasm) [reply pkg/apis/m3dboperator/v1alpha1/namespace.go]: Thanks, @schallert , `json.RawMessage` really makes the implementation much simpler. Updated.

**Review by schallert — APPROVED** [review]
LGTM, but it would be good to test on a live operated cluster before we merge, both with and without extended options. I've seen some... "unintuitive" (to put it nicely) behavior from Kubernetes before when serializing / deserializing anything non-standard.

**Comment by linasm** [comment]
> LGTM, but it would be good to test on a live operated cluster before we merge, both with and without extended options. I've seen some... "unintuitive" (to put it nicely) behavior from Kubernetes before when serializing / deserializing anything non-standard.
Sure, we've done some testing on a real cluster.

**Pull request merged**